invenio_rdm_records/collections/service.py

```diff
@@ -121,6 +127,31 @@ def add(self, identity, collection, slug, title, query, uow=None, **kwargs):
             identity, new_collection, self.collection_schema, self.links_item_tpl
         )
 
+    @unit_of_work()
+    def update(self, identity, collection_or_id, data=None, uow=None):
+        """Update a collection."""
+        if isinstance(collection_or_id, int):
+            collection = self.collection_cls.resolve(id_=collection_or_id)
+        else:
+            collection = collection_or_id
```

Comment:
just for me to understand: in what circumstances do we pass collection, not the id? I think most of the services only accept id, not the obj

Reply:
This was added to avoid resolving collections twice. I.e. this is useful when chaining service method calls. For the celery task implemented here, I need to read all the collections and then search each one of them. 

The flow of having only an `id` and resolving the entity inside works well when in a request context and when all the logic is contained within one service method.

An alternative would be having a service method that does all of that (e.g. `search_all_collections`), but it's not as granular and reusable as I would expect from a service method.